os_migrate/roles/import_workloads/tasks/workload.yml

```diff
@@ -1,133 +1,149 @@
-      ssh_key_path: "{{ os_migrate_conversion_keypair_private_path }}"
-      ssh_user: "{{ os_migrate_conversion_host_ssh_user }}"
-      transfer_uuid: "{{ exports.transfer_uuid }}"
-      src_conversion_host_address:
-        "{{ os_src_conversion_host_info.openstack_conversion_host.address }}"
-      volume_map: "{{ exports.volume_map }}"
-      log_file: "{{ prelim.log_file }}"
-      state_file: "{{ prelim.state_file }}"
-      timeout: "{{ os_migrate_timeout }}"
-    register: transfer
-    when: prelim.changed
+    - name: transfer volumes to destination
+      os_migrate.os_migrate.import_workload_transfer_volumes:
+        cloud: dst
+        validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
+        ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
+        client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
+        client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
+        data: "{{ item }}"
+        conversion_host:
+          "{{ os_dst_conversion_host_info.openstack_conversion_host }}"
+        ssh_key_path: "{{ os_migrate_conversion_keypair_private_path }}"
+        ssh_user: "{{ os_migrate_conversion_host_ssh_user }}"
+        transfer_uuid: "{{ exports.transfer_uuid }}"
+        src_conversion_host_address:
+          "{{ os_src_conversion_host_info.openstack_conversion_host.address }}"
+        volume_map: "{{ exports.volume_map }}"
+        log_file: "{{ prelim.log_file }}"
+        state_file: "{{ prelim.state_file }}"
+        timeout: "{{ os_migrate_timeout }}"
+      register: transfer
+      when:
+        - prelim is defined and prelim.changed
+        - item["_migration_params"]["data_copy"]|default(true)
 
-  - name: create destination instance
-    os_migrate.os_migrate.import_workload_create_instance:
-      cloud: dst
-      validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
-      ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
-      client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
-      client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
-      data: "{{ item }}"
-      block_device_mapping: "{{ transfer.block_device_mapping }}"
-    register: os_migrate_destination_instance
-    when: prelim.changed
+    - name: create destination instance
+      os_migrate.os_migrate.import_workload_create_instance:
+        cloud: dst
+        validate_certs: "{{ os_migrate_dst_validate_certs|default(omit) }}"
+        ca_cert: "{{ os_migrate_dst_ca_cert|default(omit) }}"
+        client_cert: "{{ os_migrate_dst_client_cert|default(omit) }}"
+        client_key: "{{ os_migrate_dst_client_key|default(omit) }}"
+        data: "{{ item }}"
+        block_device_mapping: "{{ transfer.block_device_mapping }}"
+      register: os_migrate_destination_instance
+      when:
+        - prelim is defined and prelim.changed
+        - item["_migration_params"]["data_copy"]|default(true)
```

Comment:
The e2e test path with `data_copy: false` skipped the creation of the instance because of this:

```
TASK [os_migrate.os_migrate.import_workloads : create destination instance] ****
skipping: [localhost] => changed=false 
  skip_reason: Conditional result was False
```

I think we should keep the prelim condition but only for the case when `data_copy: true` (because when running with `data_copy: false`, `prelim` will be undefined).

So i think we should have:
```
when:
  - prelim is defined and prelim.changed or not item["_migration_params"]["data_copy"]|default(true)
```

That condition should evaluate to true (task will not be skipped) when:

* item["_migration_params"]["data_copy"] is false, and then we don't care about prelim, or
* item["_migration_params"]["data_copy"] is true, and prelim is defined and it's changed.